src/bvals/comms/bnd_info.cpp

```diff
@@ -420,6 +336,15 @@ BndInfo BndInfo::GetSetBndInfo(std::shared_ptr<MeshBlock> pmb, const NeighborBlo
   } else {
     PARTHENON_FAIL("Buffer should be in a received state.");
   }
+
+  // Still don't understand why, but these have to be set
+  out.si = out.idxer[0].template StartIdx<5>();
+  out.ei = out.idxer[0].template EndIdx<5>();
+  out.sj = out.idxer[0].template StartIdx<4>();
+  out.ej = out.idxer[0].template EndIdx<4>();
+  out.sk = out.idxer[0].template StartIdx<3>();
+  out.ek = out.idxer[0].template EndIdx<3>();
```

Comment:
The prolongation/restriction machinery uses them:
```C++
GetLoopBoundsFromBndInfo(const Info_t &info, const int ckbs, const int cjbs, int &sk,
                         int &ek, int &sj, int &ej, int &si, int &ei) {
  sk = info.sk;
  ek = info.ek;
  sj = info.sj;
  ej = info.ej;
  si = info.si;
  ei = info.ei;
  if (DIM < 3) sk = ek = ckbs; // TODO(C++17) make constexpr
  if (DIM < 2) sj = ej = cjbs;
}
```
The simplest way to match the prolong/restrict machinery to your new machinery is to modify that function to use the indexer.

Reply:
Does the prolongation/restriction machinery use `BndInfo` objects that were constructed using `GetSendBndInfo` or `GetSetBndInfo`?

Reply:
> Does the prolongation/restriction machinery use BndInfo objects that were constructed using GetSendBndInfo or GetSetBndInfo?

Ah good point. No it shouldn't.

> 

I removed BndInfo::si, etc. and everything seems to be working. Still a little confused about why it mattered what I did in GetSetBndInfo, but probably not worth worrying about.

Now it uses your new machinery? Nice.